src/com/amazon/ion/system/IonReaderBuilder.java

```diff
@@ -45,23 +39,24 @@
  * in this class.
  */
 @SuppressWarnings("deprecation")
-public class IonReaderBuilder
+public abstract class IonReaderBuilder
```

Comment:
I had to extract a lot of the implementation into _Private_IonReaderBuilder to add the internal-only option to specify the LocalSymbolTableFactory, which is used by the old reader implementations.